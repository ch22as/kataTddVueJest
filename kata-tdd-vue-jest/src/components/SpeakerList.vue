<template>
  <div>SpeakerList</div>
</template>

<script>
export default {
    name: 'SpeakerList',
    props: {
        speakers: {
            type: Array,
            required: true
        }
    }
}
</script>

<style>

</style>