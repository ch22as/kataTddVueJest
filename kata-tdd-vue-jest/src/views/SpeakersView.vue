<template>
  <div class="container">

    <SearchForm/>
    <SpeakerList :speakers="speakers"/>

  </div>
</template>

<script>
import SearchForm from '@/components/SearchForm';
import SpeakerList from '@/components/SpeakerList';
import { mapGetters } from 'vuex';

export default {
    name: 'SpeakersView',
    components:{
      SearchForm,
      SpeakerList
    },
    computed: {
      ...mapGetters({speakers: 'speakers'})
    }
}
</script>

<style>

</style>