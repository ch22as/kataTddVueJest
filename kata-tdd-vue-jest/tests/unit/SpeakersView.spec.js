import { shallowMount, createLocalVue } from '@vue/test-utils';
import SpeakersView from '@/views/SpeakersView';
import SearchForm from '@/components/SearchForm.vue';
import SpeakerList from '@/components/SpeakerList.vue';
import Vuex from 'vuex';

const localVue = createLocalVue();

localVue.use(Vuex);

let speaker = {
    name: 'My Name'
};

describe('SpeakersView', () =>{

    let store, getters;

    beforeEach(() =>{
        getters = {
            speakers: () =>[speaker, speaker]
        },
        store = new Vuex.Store({ getters })
    });

    const build = () => {
        const wrapper = shallowMount(SpeakersView, {store, localVue});

        return {
            wrapper,
            SearchForm: () => wrapper.findComponent(SearchForm),
            SpeakerList: () => wrapper.findComponent(SpeakerList)
        }
    }
    //Existe el componente segÃºn el snapshots?
    it('it renders the component', () =>{
        const wrapper = shallowMount(SpeakersView);

        expect(wrapper.html()).toMatchSnapshot();
    });

    //EstÃ¡ los componentes en SpeakersView?
    it('it renders the right', () =>{
       const { SearchForm, SpeakerList} = build();

        expect(SearchForm().exists()).toBe(true);
        expect(SpeakerList().exists()).toBe(true);   
    });

    it('it passes speakers to the speakerList', () =>{
        const { wrapper, SpeakerList} = build();

        expect(SpeakerList().vm.speakers).toEqual([speaker, speaker])
        expect(SpeakerList().vm.speakers).toBe(store.getters.speakers)
    })
});