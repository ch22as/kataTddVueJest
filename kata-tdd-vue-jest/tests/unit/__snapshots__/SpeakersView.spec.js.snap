// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpeakersView it renders the component 1`] = `
<div class="container">
  <searchform-stub></searchform-stub>
  <speakerlist-stub></speakerlist-stub>
</div>
`;
